// @paljs/ui/Popover/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = require("styled-components");

var _style = _interopRequireDefault(require("./style"));

var _PopoverLay = _interopRequireDefault(require("../PopoverLay"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/*
 * @license
 * Copyright Ahmed Elywa. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
var _ref = /*#__PURE__*/_react["default"].createElement("span", {
  className: "arrow"
});

var Popover = function Popover(props) {
  var theme = _react["default"].useContext(_styledComponents.ThemeContext);

  var arrowSize = parseInt(theme.popoverArrowSize);
  var arrowRound = Math.round(-arrowSize - arrowSize / 2);
  return /*#__PURE__*/_react["default"].createElement(_PopoverLay["default"], {
    target: props.children,
    placement: props.placement,
    trigger: props.trigger,
    style: props.style,
    className: props.className,
    eventListener: props.eventListener,
    transformSize: 15,
    arrowRound: arrowRound,
    arrowSize: theme.popoverArrowSize
  }, /*#__PURE__*/_react["default"].createElement(_style["default"], null, _ref, typeof props.overlay === 'string' ? /*#__PURE__*/_react["default"].createElement("div", {
    className: "primitive-overlay"
  }, props.overlay) : props.overlay));
};

var _default = Popover;
exports["default"] = _default;
