// @paljs/ui/Checkbox/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Checkbox = void 0;

var _react = _interopRequireDefault(require("react"));

var _style = _interopRequireDefault(require("./style"));

var _Icon = require("../Icon");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/*
 * @license
 * Copyright Ahmed Elywa. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
var _ref = /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
  name: "checkmark-bold-outline"
});

var _ref2 = /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
  name: "minus-bold-outline"
});

var Checkbox = function Checkbox(props) {
  return /*#__PURE__*/_react["default"].createElement(_style["default"], {
    disabled: props.disabled,
    checked: props.checked,
    status: props.status,
    className: props.className,
    style: props.style
  }, /*#__PURE__*/_react["default"].createElement("input", {
    type: "checkbox",
    className: "native-input visually-hidden",
    disabled: props.disabled,
    onBlur: props.onBlur,
    checked: props.checked,
    onChange: function onChange(e) {
      return props.onChange(e.target.checked);
    }
  }), /*#__PURE__*/_react["default"].createElement("span", {
    className: "custom-checkbox"
  }, props.checked && !props.indeterminate && _ref, props.indeterminate && _ref2), /*#__PURE__*/_react["default"].createElement("span", {
    className: "text"
  }, props.children));
};

exports.Checkbox = Checkbox;
Checkbox.defaultProps = {
  status: 'Basic'
};
